code/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/forms/CalendarFieldForm.java

```diff
@@ -302,29 +365,60 @@ protected Set<? extends IMenuType> getConfiguredMenuTypes() {
           protected void execAction() {
             CalendarComponentForm form = new CalendarComponentForm();
             form.start();
-            if (getCalendar().getSelectedRange() != null) {
-              form.getStartDateField().setValue(getCalendar().getSelectedRange().getFrom());
-              form.getEndDateField().setValue(getCalendar().getSelectedRange().getTo());
+            form.setCalendars(getCalendars());
+            if (getSelectedRange() != null) {
+              form.getStartDateField().setValue(getSelectedRange().getFrom());
+              form.getEndDateField().setValue(getSelectedRange().getTo());
+            }
+            if (getSelectedCalendar() != null) {
+              form.getCalendarField().setValue(getSelectedCalendar());
             }
             form.waitFor();
             if (form.isFormStored()) {
-              String cssClass = "calendar-appointment";
               Date start = form.getStartDateField().getValue();
               Date end = form.getEndDateField().getValue();
-              ICalendarAppointment calendarAppointment = new CalendarAppointment(0L, 0L, start, end, false, null, form.getSubjectField().getValue(), null, cssClass);
+              boolean fullDay = form.getFullDayField().getValue();
+              ICalendarAppointment calendarAppointment = new CalendarAppointment(0L, 0L, start, end, fullDay, null, form.getSubjectField().getValue(), null, null);
               calendarAppointment.setSubjectLabel("Appointment");
               if (form.getLocationField().getValue() != null) {
                 calendarAppointment.getDescriptionElements().add(BEANS.get(CalendarItemDescriptionElement.class).withText(form.getLocationField().getValue()).withIconId(Icons.World));
               }
               if (form.getDescriptionField().getValue() != null) {
                 calendarAppointment.getDescriptionElements().add(BEANS.get(CalendarItemDescriptionElement.class).withText(form.getDescriptionField().getValue()).withIconId(Icons.List));
               }
+              if (form.getCalendarField().getValue() != null) {
+                calendarAppointment.setCalendarId(form.getCalendarField().getValue().getCalendarId());
+              }
               m_configuredAppointments.add(calendarAppointment);
               getCalendar().reloadCalendarItems();
               getCalendar().setSelectedRange(null);
             }
           }
         }
+
+        @Order(2000)
+        @ClassId("0918cbdd-c96d-4f4e-b280-1490ec28ad68")
+        public class AddCalendarMenu extends AbstractMenu {
+          @Override
+          protected String getConfiguredText() {
+            return "Add calendar";
+          }
+
+          @Override
+          protected Set<? extends IMenuType> getConfiguredMenuTypes() {
+            return CollectionUtility.hashSet(CalendarMenuType.EmptySpace);
```

Comment:
Not Calendar Menu Type, because should not appear as right click on the calendar, but as form menu